### PR TITLE
feat: slider: enable basic data comparison slider implementations

### PR DESCRIPTION
**Pull request by dkilgore-eightfold**
## SUMMARY:
Enables Data Comparison implementations of `Slider` and extends customizability.

### Bug fixes:

- Fixes thumb value position with custom inline extremity label offset and marker offset via `thumbGeometry` function and updated logic in `updateLayout`
- Fixes thumb cursor to use `gipper` and `gripping`
- Exports `SliderSize` enum
- Fixes marks prop by exposing full `SliderMarker` for internal use
- Adds some typing
- Add `readOnly` to `<input>` if `Slider` is `readOnly`
- Adds `key` to `Tooltip`
- Ensure `changeToCloseValue` function handling when `readOnly`

### Features:

- Adds Benchmark Slider Story
- Adds Data Inactive Story
- Adds Data Active Story
- Extends varTheme support of `Slider`
- Demonstrates how to use SVGs as markers
- Provides Data Comparison SVGs for quick reference
- Adds Ability to hide the `Slider` track and rail via `hideTrack` and `hideRail` props
- Adds ability to quickly update the track color of `Slider` via `trackColor` prop `SliderTrackColor` enum
- Adds ability to give `Slider` `dots` custom classes via `dotClassNames` prop

https://user-images.githubusercontent.com/99700808/216201856-16e2527e-0f1d-4981-af9f-4b00314a72c9.mp4

## JIRA TASK (Eightfold Employees Only):
ENG-32662

## CHANGE TYPE:

- [X] Bugfix Pull Request
- [X] Feature Pull Request

## TEST COVERAGE:

- [ ] Tests for this change already exist
- [X] I have added unittests for this change

## TEST PLAN:
Pull the PR branch and run `yarn` and `yarn storybook`. Verify the `Slider` stories behave as expected.

### CODE REVIEW

File: src/components/Slider/slider.module.scss [hunk 1]
```diff
@@ -284,21 +347,25 @@ $small-marker-top: 0;
     // For Chrome browsers
     &::-webkit-slider-thumb {
       -webkit-appearance: none;
-      background-color: var(--slider-default-thumb-color);
+      background-color: var(--slider-thumb-color);
```
Comment (jhoward-eightfold) [inline src/components/Slider/slider.module.scss:350]: highly unlikely anyone is actually using this yet, but are we considering some of these var name adjustments breaking change? in the future we definitely should

File: src/styles/themes/_default-theme.scss [hunk 2]
```diff
@@ -712,4 +712,33 @@
...
+  --slider-rail-border-color: var(--primary-color-60);
+  --slider-thumb-color: var(--accent-color-20);
+  --slider-track-background-color: var(--primary-color-60);
+  --slider-track-border-color: var(--primary-color-60);
+  --slider-track-green-background-color: var(--green-color-30);
+  --slider-track-green-border-color: var(--green-color-60);
+  --slider-track-orange-background-color: var(--orange-color-30);
+  --slider-track-orange-border-color: var(--orange-color-60);
+  --slider-track-red-background-color: var(--red-color-30);
+  --slider-track-red-border-color: var(--red-color-60);
```
Comment (jhoward-eightfold) [inline src/styles/themes/_default-theme.scss:741]: recommend naming these according to status rather than color
slider-track-success, warning, error

**Review by jhoward-eightfold — APPROVED** [review]
approved with comments

File: src/components/Slider/slider.module.scss (hunk 1 above)
Comment (dkilgore-eightfold) [reply src/components/Slider/slider.module.scss:350]: Agreed, figure it's early enough from the outset of these var names to update them as all forks of the repo are Eightfold folks.

Comment (dkilgore-eightfold) [reply src/components/Slider/slider.module.scss:350]: Given that was a scoped variable (not part of varTheme), we should be okay.

**Pull request merged**